pkg/controllers/state/cluster.go

```diff
@@ -82,6 +82,16 @@ func NewCluster(clk clock.Clock, client client.Client) *Cluster {
 //
 //nolint:gocyclo
 func (c *Cluster) Synced(ctx context.Context) (synced bool) {
+	start := time.Now()
```

Comment:
This isn't going to be held onto through multiple calls to this function. You need to retain the information in the cluster state of the first time that we were unsynced and then diff from that